pkg/cmd/workflow/view/view.go

```diff
@@ -135,19 +142,12 @@ func runView(opts *ViewOptions) error {
 	return nil
 }
 
-func viewWorkflowContent(opts *ViewOptions, client *api.Client, workflow *shared.Workflow) error {
-	repo, err := opts.BaseRepo()
-	if err != nil {
-		return fmt.Errorf("could not determine base repo: %w", err)
-	}
-
-	opts.IO.StartProgressIndicator()
-	yamlBytes, err := shared.GetWorkflowContent(client, repo, *workflow, opts.Ref)
-	opts.IO.StopProgressIndicator()
+func viewWorkflowContent(opts *ViewOptions, client *api.Client, repo ghrepo.Interface, workflow *shared.Workflow, ref string) error {
+	yamlBytes, err := shared.GetWorkflowContent(client, repo, *workflow, ref)
 	if err != nil {
 		if s, ok := err.(api.HTTPError); ok && s.StatusCode == 404 {
-			if opts.Ref != "" {
-				return fmt.Errorf("could not find workflow file %s on %s, try specifying a different ref", workflow.Base(), opts.Ref)
+			if ref != "" {
```

Comment:
Is there a reason to explicitly pass in `ref` here? Seems like we are still passing in `opts`.

Reply:
That's a valid observation, but I felt it was much more clear for the signature of the method to receive the ref as argument rather than inferring it from `opts`. Furthermore, I feel like too having many methods having `opts` as a receiver or argument is an antipattern because it dis-incentivizes the method from having a very limited responsibility. This small refactoring was a step towards not passing `opts` as an argument in the future.